### PR TITLE
deposit form: add references field to deposit form

### PR DESCRIPTION
**Pull request by mitsosf**
* closes https://github.com/inveniosoftware/invenio-app-rdm/issues/2169

### Description

Adds references field to deposit form

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [x] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [x] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [x] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).

### CODE REVIEW

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/ReferencesField/initialValues.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// This file is part of Invenio-RDM-Records
+// Copyright (C) 2020-2023 CERN.
+// Copyright (C) 2020-2022 Northwestern University.
+//
+// Invenio-RDM-Records is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+export const emtpyReference = {
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/ReferencesField/initialValues.js:7]:
```suggestion
export const emptyReference = {
```

File: invenio_rdm_records/config.py [hunk 2]
```diff
@@ -63,16 +63,23 @@ def always_valid(identifier):
         "validator": idutils.is_ads,
         "datacite": "bibcode",
     },
+    "crossreffunderid": {
```
Comment (kpsherva) [inline invenio_rdm_records/config.py:66]: ping @slint do we want to merge these variables?

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/ReferencesField/ReferencesField.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+
+import {
+  TextField,
+  GroupField,
+  ArrayField,
+  FieldLabel,
+  SelectField,
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/ReferencesField/ReferencesField.js:17]: SelectField seems unused, could be removed

File: invenio_rdm_records/config.py (hunk 2 above)
Comment (slint) [reply invenio_rdm_records/config.py:66]: I think it's fine, references are kind of like `related_identifiers` so they should both accept the same types of identifiers.

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/ReferencesField/ReferencesField.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  label: PropTypes.string,
+  labelIcon: PropTypes.string,
+  required: PropTypes.bool,
+  showEmptyValue: PropTypes.bool,
+};
+
+ReferencesField.defaultProps = {
+  label: i18next.t("References"),
+  labelIcon: "bookmark",
+  required: undefined,
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/ReferencesField/ReferencesField.js:74]:
```suggestion
  required: false,
```

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/ReferencesField/initialValues.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+// This file is part of Invenio-RDM-Records
+// Copyright (C) 2020-2023 CERN.
+// Copyright (C) 2020-2022 Northwestern University.
+//
+// Invenio-RDM-Records is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+export const emptyReference = {
+  reference: "",
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/ReferencesField/initialValues.js:8]:
```suggestion
  reference: {},
```

**Review by kpsherva — APPROVED** [review]

**Pull request merged**